pkg/dockerregistry/server/admin.go

```diff
@@ -112,5 +121,81 @@ func (bh *blobHandler) Delete(w http.ResponseWriter, req *http.Request) {
 		dcontext.GetLogger(bh).Infof("blobHandler: ignoring %T error: %v", err, err)
 	}
 
+	query := req.URL.Query()
+
+	if len(query.Get("forwarded")) == 0 {
+		go bh.propagateRequest(*req.URL)
+	}
+
 	w.WriteHeader(http.StatusNoContent)
 }
+
+func (bh *blobHandler) propagateRequest(rurl url.URL) {
+	registryClient := &http.Client{
+		Transport: http.DefaultTransport,
+	}
+
+	if rurl.Scheme == "https" {
+		registryClient.Transport, _ = restclient.TransportFor(&restclient.Config{
+			TLSClientConfig: restclient.TLSClientConfig{Insecure: true},
+		})
+	}
+
+	var err error
+
+	hostname := os.Getenv("HOSTNAME")
+	if len(hostname) == 0 {
+		hostname, err = os.Hostname()
+		if err != nil {
+			dcontext.GetLogger(bh).Errorf("propagateRequest: hostname error: %s", err)
+			return
+		}
+	}
+
+	hostAddrs, err := net.LookupHost(hostname)
+	if err != nil {
+		dcontext.GetLogger(bh).Errorf("propagateRequest: lookup hostname error: %s", err)
+		return
+	}
+
+	addrs, err := net.LookupHost(pruneService)
```

Comment:
what is creating prune-service?  I assume it's intended to be a headless service so you can get all the addressed of the endpoints?

Reply:
> what is creating prune-service?

Right now no one. But if we are going to merge this, then we can create it in the `cluster-image-registry-oprator`.

> I assume it's intended to be a headless service so you can get all the addressed of the endpoints?

Yes. I want to have another headless service with our selector to get dns records for all endpoints.